### PR TITLE
LPS-174358 FDS | Tab key navigation skips dropdown on filter label

### PR DESCRIPTION
**Pull request by kresimir-coko**
https://issues.liferay.com/browse/LPS-174358

In order to have the proper focus behaviour it was needed to pass a `ClayButton` component instead of the `ClayLabel` component as a `trigger` for the `ClayDropDown`.

I have also noticed a lot of SF inconsistencies in the `frontend-data-set-web` module, for example missing empty lines between sibling elements. Are we disabling our ESLint rules somehow in this module?

### CODE REVIEW

**Review by markocikos — APPROVED** [review]
LGTM! :sparkles: 

SF error looks unrelated.

I typically make sure to have `FDS` in PR title, so I can easily search for relevant closed PRs on GitHub.

**Comment by kresimir-coko** [comment]
What about this I posted in the initial comment @markocikos ?
> I have also noticed a lot of SF inconsistencies in the frontend-data-set-web module, for example missing empty lines between sibling elements. Are we disabling our ESLint rules somehow in this module?

**Comment by markocikos** [comment]
> I have also noticed a lot of SF inconsistencies in the frontend-data-set-web module, for example missing empty lines between sibling elements. Are we disabling our ESLint rules somehow in this module?

AFAIK, we don't have a special ESLint rule for this. It is a rule for JSP tags, enforced by `sourceFormat` script. JSP tags have similar syntax to JSX, so I like to mirror the same rule. But no, we don't formally apply this to JS files. Both blank row and no rows are allowed between siblings.

**Comment by kresimir-coko** [comment]
I found out the exact behaviour and reported a [bug to frontend-projects](https://github.com/liferay/liferay-frontend-projects/issues/1098)

**Comment by kresimir-coko** [comment]
ci:reevaluate:1397401_3430

**Comment by markocikos** [comment]
Failures are repeating, may be related.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/FilterResume.js [hunk 1]
```diff
@@ -28,34 +28,15 @@ function FilterResume(props) {
...
 	const [open, setOpen] = useState(false);
 
-	const label = (
-		<ClayLabel
+	const button = (
+		<ClayButton
 			className={classNames(
-				'filter-resume component-label tbar-label mr-2',
+				'filter-resume component-label tbar-label',
 				props.disabled && 'disabled',
```
Comment (carloslancha) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/FilterResume.js:35]: Shouldn't this be passed as `disabled` param instead as a class?

**Comment by kresimir-coko** [comment]
I think QA needs to update tests in order to get this green, as the tests are expecting a `label` class which is not a `btn`

**Comment by mateusralv** [comment]
Hi @kresimir-coko yes, I just found a related bug, I will change the path and make a commit, I don't have permission to add it here, so I will make a new PR and could you add it here?

**Comment by mateusralv** [comment]
Hi @kresimir-coko I will create a ticket for the test fix to be done later so this PR will be forwarded faster. So the failures are not related, seems safe forward. 

Test that needs to be fixed: FDSFilters#OneFilterCanBeRemoved

[Testray comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1678372363984&testrayRunIdA=2225028894&testrayRunIdB=2225320760)

**Comment by kresimir-coko** [comment]
Manually forwarded to https://github.com/brianchandotcom/liferay-portal/pull/131463

**Comment by markocikos** [comment]
@kresimir-coko @mateusralv Sorry, but I closed the forwarded PR on Brian's fork. Is this really safe to manually forward? There is a legitimate failure in `master`. If the manually forwarded PR was merged, wouldn't it cause CI failures on all PRs related to FDS? Shouldn't we fix the test first, or at least quarantine it? /cc @manoelcyreno

**Comment by kresimir-coko** [comment]
Same concern that I had but Mateus' wording made it seem safe to forward 🤷

**Comment by mateusralv** [comment]
Hi @markocikos yes most of the time we send the test fix along with the PR. In this case, I noticed that the failure occurs in only 1 FDS test, and we thought that this way would be faster and we could solve it after the PR is merged. But I can update this PR with the Fix if your prefer. What do you think? 
Cc: @manoelcyreno

**Comment by manoelcyreno** [comment]
@mateusralv , can you update this PR, putting the test in "ignore" and adding a comment to fix this test in another ticket?
https://issues.liferay.com/browse/LPS-178165 - Fix the poshi test impacted by the PR liferay-frontend#3094

**Comment by kresimir-coko** [comment]
Closing in favor of https://github.com/liferay-frontend/liferay-portal/pull/3115 which contains test updates, too

**Pull request closed without merging**